kernel/names.mli

```diff
@@ -549,17 +549,72 @@ type module_path = ModPath.t =
 [@@ocaml.deprecated "Alias type"]
 
 module Projection : sig
-  type t
+  module Repr : sig
+    type t = private
+      { proj_ind : inductive;
+        proj_npars : int; (** Number of parameters. *)
+        proj_arg : int; (** Index of the projection, 0-based and ignoring parameters. *)
+        proj_name : Label.t; }
+
+    val make : inductive -> proj_npars:int -> proj_arg:int -> Label.t -> t
```

Comment:
Why make the record private and redefine it constructor?

Reply:
I do prefer providing abstract builders, this allows for future modification of this code without messing to much with the callers.

Reply:
We've had this discussion before, but I generally don't buy the arguments, unless it is meant to enforce some invariant. What is the concrete future modification scenario where it would help?

The codebase is bloated with useless aliases, we shouldn't introduce more without reasons. If there are some reasons, then I can hear them.